### PR TITLE
SF-1192: Do not show SAYT if search isFetching.

### PR DESCRIPTION
**Pull request by JoeyDeol**

### CODE REVIEW

File: packages/@storefront/flux-capacitor/src/core/reducers/is-fetching.ts [hunk 1]
```diff
@@ -39,7 +39,11 @@ export default function updateIsFetching(state: State = DEFAULT_FETCHING, action
     case Actions.RECEIVE_AUTOCOMPLETE_SUGGESTIONS: return doneFetching(state, 'autocompleteSuggestions');
     case Actions.FETCH_AUTOCOMPLETE_PRODUCTS: return startFetching(state, 'autocompleteProducts');
     case Actions.RECEIVE_AUTOCOMPLETE_PRODUCTS: return doneFetching(state, 'autocompleteProducts');
-    case Actions.FETCH_PRODUCT_DETAILS: return startFetching(state, 'details');
+    case Actions.FETCH_PRODUCT_DETAILS:
+      if (action.payload.redirect) {
```
Comment (jmbtutor) [inline packages/@storefront/flux-capacitor/src/core/reducers/is-fetching.ts]: Maybe this is better as a ternary?

File: packages/@storefront/flux-capacitor/src/core/selectors.ts [hunk 2]
```diff
@@ -363,6 +363,21 @@ namespace Selectors {
   export const config = (state: Store.State) =>
     state.session.config;
 
+  /**
+   * If called with a `value` parameter, returns a boolean value indicating the
+   * current fetching state otherwise returns the full `isFetching` object.
+   * @param value Accepted values conform to the [[Store.IsFetching]] properties.
```
Comment (jmbtutor) [inline packages/@storefront/flux-capacitor/src/core/selectors.ts]: Needs an `@return` line.

File: packages/@storefront/flux-capacitor/src/core/selectors.ts [hunk 3]
```diff
@@ -363,6 +363,21 @@ namespace Selectors {
...
+   */
+  export const isFetching = (state: Store.State, value?: keyof Store.IsFetching) => {
+    if (value && value in state.isFetching) {
```
Comment (jmbtutor) [inline packages/@storefront/flux-capacitor/src/core/selectors.ts]: Unless TypeScript needs it, checking if `value` is truthy is redundant.

File: packages/@storefront/sayt/test/unit/autocomplete.ts [hunk 4]
```diff
@@ -313,6 +314,24 @@ suite('Autocomplete', ({ expect, spy, stub, itShouldProvideAlias }) => {
 
       expect(emit).to.be.calledWith('sayt:show');
     });
+
+    it('should do nothing when search isFetching', () => {
+      const emit = spy();
+      const setActivation = autocomplete.setActivation = spy();
+      select = Autocomplete.prototype.select = <any>spy(() => true);
```
Comment (jmbtutor) [inline packages/@storefront/sayt/test/unit/autocomplete.ts]: Does this need to be cast?

**Comment by jmbtutor** [comment]
Also make sure you make an entry in the changelog

File: packages/@storefront/flux-capacitor/src/core/selectors.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (jmbtutor) [reply packages/@storefront/flux-capacitor/src/core/selectors.ts]: re `Object.keys(state.isFetching).includes(value);`: `Object.prototype.hasOwnProperty()` exists for this purpose.

File: packages/@storefront/flux-capacitor/src/core/selectors.ts [hunk 5]
```diff
@@ -363,6 +363,21 @@ namespace Selectors {
   export const config = (state: Store.State) =>
     state.session.config;
 
+  /**
+   * If called with a `value` parameter, returns a boolean value indicating the
```
[the remark this answers is not in the document]
Comment (Vic-Dev) [reply packages/@storefront/flux-capacitor/src/core/selectors.ts]: I thought this was describing the most common use case first, but if you just want the order reversed, will do 👍

File: packages/@storefront/flux-capacitor/src/core/selectors.ts (hunk 2 above)
Comment (Vic-Dev) [reply packages/@storefront/flux-capacitor/src/core/selectors.ts]: No, it's auto-generated

Comment (Vic-Dev) [reply packages/@storefront/flux-capacitor/src/core/selectors.ts]: We just added to the auto-generated param definition bc it doesn't link the type

Comment (jmbtutor) [reply packages/@storefront/flux-capacitor/src/core/selectors.ts]: It's more for the English explanation for what it returns.

Comment (Vic-Dev) [reply packages/@storefront/flux-capacitor/src/core/selectors.ts]: Ok but we don't need it, so no

File: packages/@storefront/flux-capacitor/CHANGELOG.md [hunk 6]
```diff
@@ -4,6 +4,15 @@ All notable changes to this project will be documented in this file.
...
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased] [<release type>]
+### Changed
+- SF-1192: Update `fetchProductDetails` action to include redirect parameter.
+  - Set `search` property of `isFetching` state slice within the store to `false` when a single result redirect is triggered.
+
+### Added
+- SF-1192: Create isFetching selector
+  - Selector is needed in order to modify and track isFetching assocaited portion of the store.
```
Comment (Vic-Dev) [inline packages/@storefront/flux-capacitor/CHANGELOG.md]: `associated`

File: packages/@storefront/flux-capacitor/CHANGELOG.md [hunk 7]
```diff
@@ -4,6 +4,15 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased] [<release type>]
+### Changed
+- SF-1192: Update `fetchProductDetails` action to include redirect parameter.
+  - Set `search` property of `isFetching` state slice within the store to `false` when a single result redirect is triggered.
+
+### Added
+- SF-1192: Create isFetching selector
```
Comment (Vic-Dev) [inline packages/@storefront/flux-capacitor/CHANGELOG.md]: `isFetching` use back ticks

File: packages/@storefront/flux-capacitor/CHANGELOG.md (hunk 6 above)
Comment (Vic-Dev) [reply packages/@storefront/flux-capacitor/CHANGELOG.md]: Actually, selectors are not used to modify, so just `to retrieve isFetching portion of the store`

File: packages/@storefront/flux-capacitor/CHANGELOG.md [hunk 8]
```diff
@@ -4,6 +4,15 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased] [<release type>]
```
Comment (jmbtutor) [inline packages/@storefront/flux-capacitor/CHANGELOG.md]: Make sure to add the proper release type.

File: packages/@storefront/flux-capacitor/CHANGELOG.md (hunk 7 above)
Comment (jmbtutor) [inline packages/@storefront/flux-capacitor/CHANGELOG.md]: Add backticks around `isFetching`.

File: packages/@storefront/flux-capacitor/CHANGELOG.md (hunk 6 above)
Comment (jmbtutor) [inline packages/@storefront/flux-capacitor/CHANGELOG.md]: Add backticks around `isFetching`.

File: packages/@storefront/sayt/CHANGELOG.md [hunk 9]
```diff
@@ -4,6 +4,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased] [<release type>]
+### Changed
```
Comment (Vic-Dev) [inline packages/@storefront/sayt/CHANGELOG.md]: I think this is `Fixed` not `Changed` but I could be wrong?

File: packages/@storefront/sayt/CHANGELOG.md [hunk 10]
```diff
@@ -4,6 +4,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased] [<release type>]
```
Comment (jmbtutor) [inline packages/@storefront/sayt/CHANGELOG.md]: Make sure to add the proper release type.

File: packages/@storefront/sayt/CHANGELOG.md (hunk 9 above)
Comment (jmbtutor) [reply packages/@storefront/sayt/CHANGELOG.md]: I agree.

File: packages/@storefront/sayt/CHANGELOG.md [hunk 11]
```diff
@@ -4,6 +4,11 @@ All notable changes to this project will be documented in this file.
...
+### Changed
+- SF-1192: Only show sayt if search has finished fetching.
+  - While search is fetching, set sayt to inactive.
```
Comment (jmbtutor) [inline packages/@storefront/sayt/CHANGELOG.md]: Maybe add a brief explanation of the behaviour it fixed if you changed it to `fixed`.

**Pull request merged**